Library/Homebrew/download_strategy.rb

```diff
@@ -1148,6 +1142,9 @@ def self.detect_from_url(url)
       SubversionDownloadStrategy
     when %r{^https?://(.+?\.)?sourceforge\.net/hgweb/}
       MercurialDownloadStrategy
+    when %r{^s3://}
+      require_aws_sdk
```

Comment:
Why is this needed in two places?

Reply:
This is necessary to support both cases when the formula explicitly states `:using => S3DownloadStrategy` and again when it’s implied by using the `s3://` scheme in the download URL.

Ex:

```
# Traditional URL with :using
url “https://but.s3.anazonaws.com/key” :using => S3DownloadStrategy

# S3 URL
url “s3://bkt/key”
```

Reply:
As noted above, I can separate these into different PRs, but I think it’s needed in both places

Reply:
As noted above, I can separate these into different PRs, but I think it’s needed in both places

Reply:
It would be good to have:

1. the gem changes to the existing code
2. the `s3://` extensions

be separate PRs, yeh.